src/main/java/org/libdohj/names/NameLookupLatestRestHeightApi.java

```diff
@@ -0,0 +1,124 @@
+package org.libdohj.names;
+
+import org.bitcoinj.core.BlockChain;
+import org.bitcoinj.core.Transaction;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+// TODO: document this
+
+public class NameLookupLatestRestHeightApi implements NameLookupLatest {
+    
+    protected BlockChain chain;
+    protected NameLookupByBlockHeight heightLookup;
+    protected String restUrlPrefix;
+    protected String restUrlSuffix;
+    
+    public NameLookupLatestRestHeightApi (String restUrlPrefix, String restUrlSuffix, BlockChain chain, NameLookupByBlockHeight heightLookup) {
+        this.restUrlPrefix = restUrlPrefix;
+        this.restUrlSuffix = restUrlSuffix;
+        this.chain = chain;
+        this.heightLookup = heightLookup;
+    }
+    
+    // TODO: make a new Exception class
+    public Transaction getNameTransaction(String name, String identity) throws Exception {
```

Comment:
You should probably have an `@Override` tag on this method since it implements the method in the interface.